src/Acuminator/Acuminator.Analyzers/StaticAnalysis/BqlParameterMismatch/Helpers/ParametersCounterSyntaxWalker.cs

```diff
@@ -54,12 +54,25 @@ public override void VisitGenericName(GenericNameSyntax genericNode)
 				if (genericNode.IsUnboundGenericName)
 					typeSymbol = typeSymbol.OriginalDefinition;
 
-				if (!ParametersCounter.CountParametersInTypeSymbol(typeSymbol, _cancellationToken))
+				if (!ParametersCounter.CountParametersInTypeSymbolForGenericNode(typeSymbol, _cancellationToken))
 					return;
 
 				if (!_cancellationToken.IsCancellationRequested)
 					base.VisitGenericName(genericNode);
 			}
+
+			public override void VisitIdentifierName(IdentifierNameSyntax identifierNode)
+			{
+				if (_cancellationToken.IsCancellationRequested)
```

Comment:
Why not `ThrowIfCancellationRequested`? I may be wrong but it seems that the walker can finish its work with unexpected results instead of global cancellation of the whole process.

Reply:
Yes, it's quite old diagnostic, we certainly can do ThrowIfCancellationRequested. I'll change it

Reply:
Done